File: src/python/core/project_paths.py
from __future__ import annotations

from pathlib import Path


PROJECT_ROOT = Path(__file__).resolve().parents[3]
SRC_ROOT = PROJECT_ROOT / "src"
PYTHON_ROOT = SRC_ROOT / "python"
TS_ROOT = SRC_ROOT / "ts"
FRONTEND_ROOT = SRC_ROOT / "frontend"
FRONTEND_DIST_DIR = FRONTEND_ROOT / "dist"
CONFIG_PATH = PROJECT_ROOT / "config.json"
SESSIONS_DIR = PROJECT_ROOT / "sessions"
OUTPUT_DIR = PROJECT_ROOT / "output"
AGENT_WORKSPACE_DIR = PROJECT_ROOT / "agent_workspace"
AGENT_MEDIA_DIR = AGENT_WORKSPACE_DIR / "media"
TMP_DIR = PROJECT_ROOT / ".tmp"


__all__ = [
    "PROJECT_ROOT",
    "SRC_ROOT",
    "PYTHON_ROOT",
    "TS_ROOT",
    "FRONTEND_ROOT",
    "FRONTEND_DIST_DIR",
    "CONFIG_PATH",
    "SESSIONS_DIR",
    "OUTPUT_DIR",
    "AGENT_WORKSPACE_DIR",
    "AGENT_MEDIA_DIR",
    "TMP_DIR",
]

File: src/ts/src/core/project-paths.ts
import path from "node:path";

export function getTsProjectRoot(): string {
  return process.env.TS_AGENT_PROJECT_ROOT?.trim() || process.cwd();
}

export function getProjectRoot(): string {
  return process.env.DOC_FORMAT_PROJECT_ROOT?.trim() || path.resolve(getTsProjectRoot(), "..", "..");
}

export function getSessionsDir(): string {
  return path.join(getProjectRoot(), "sessions");
}

export function getOutputDir(): string {
  return path.join(getProjectRoot(), "output");
}

export function getAgentWorkspaceDir(): string {
  return path.join(getProjectRoot(), "agent_workspace");
}

export function getAgentMediaDir(): string {
  return path.join(getAgentWorkspaceDir(), "media");
}

export function getTempDir(): string {
  return path.join(getProjectRoot(), ".tmp");
}

export function getPythonToolRunnerPath(): string {
  return path.join(getProjectRoot(), "src", "python", "api", "python_tool_runner.py");
}

export function getWriterScriptPath(): string {
  return path.join(getProjectRoot(), "scripts", "write_docx_from_ir.py");
}
